_episodes/08-edge-detection.md

```diff
@@ -49,430 +45,376 @@ black and and one white:
+high_threshold = float(sys.argv[4])
 ~~~
-{: .pythpn}
+{: .python}
 
-Next, the original images is read, in grayscale, and displayed. 
+Next, the original images is read, in grayscale, and displayed.
 
 ~~~
 # load and display original image as grayscale
-image = cv2.imread(filename = filename, flags = cv2.IMREAD_GRAYSCALE)
-cv2.namedWindow(winname = "original", flags = cv2.WINDOW_NORMAL)
-cv2.imshow(winname = "original", mat = image)
-cv2.waitKey(delay = 0)
+image = skimage.io.imread(fname=filename, as_gray=True)
+viewer = skimage.viewer(image=image)
+viewer.show()
 ~~~
 {: .python}
 
 Then, we apply Canny edge detection with this function call:
 
 ~~~
-edges = cv2.Canny(image = image, 
-    threshold1 = lo,
-    threshold2 = hi)
+edges = skimage.feature.canny(
+    image=image,
+    sigma=sigma,
+    low_threshold=low_threshold,
+    high_threshold=high_threshold,
+)
 ~~~
 {: .python}
 
-As we are using it here, the `cv2.Canny()` function takes three parameters. The
-first parameter is the input image. The next two parameters are the low and 
-high threshold values for the fourth step of the process. 
+As we are using it here, the `skimage.feature.canny()` function takes four parameters.
+The first parameter is the input image. The `sigma` parameter determines the
+amount of Gaussian smoothing that is applied to the image. The next two
+parameters are the low and high threshold values for the fourth step of the
+process.
 
-The result of this call is a binary image. In the image, the edges detected by 
-the process are white, while everything else is black. 
+The result of this call is a binary image. In the image, the edges detected by
+the process are white, while everything else is black.
 
-Finally, the program displays the `edges` image, showing the edges that were 
-found in the original. 
+Finally, the program displays the `edges` image, showing the edges that were
+found in the original.
 
 ~~~
 # display edges
-cv2.namedWindow(winname = "edges", flags = cv2.WINDOW_NORMAL)
-cv2.imshow(winname = "edges", mat = edges)
-cv2.waitKey(delay = 0)
+viewer = skimage.viewer.ImageViewer(edges)
+viewer.show()
 ~~~
 {: .python}
 
 Here is the result, for the colored shape image above,
-with low threshold value 35 and high threshold value 55:
+with sigma value 2.0, low threshold value 0.1 and high threshold value 0.3:
 
-![Output file of Canny edge detection](../fig/07-canny-edges.jpg)
+![Output file of Canny edge detection](../fig/07-canny-edges.png)
 
-Note that the edge output shown in an OpenCV window may look significantly
+Note that the edge output shown in an skimage window may look significantly
 worse than the image would look if it were saved to a file. The image above
-is the edges of the junk image, saved in a JPG file. Here is how the same 
-image looks when displayed in an OpenCV output window:
+is the edges of the junk image, saved in a PNG file. Here is how the same
+image looks when displayed in an skimage output window:
```

Comment:
This is a bit confusing to me? Why does the image in the viewer look worse? Is this some sampling artifact when resizing the image in the viewer? Also, the image still shows the OpenCV viewer, right?

Reply:
i assume it's the resampling, the viewer shown should be skimage, but I'll check later whether I forgot to check it in (looked really bad there, too)

Reply:
Ok, maybe let's add a sentence that says it's due to resampling artefacts?
Of course, this opens up a whole new chapter of thinks that we can't really discuss here,
but I fell that without any explanation it is even weirder.

I think you're right, the image shows the skimage viewer, you have changed the file.